### PR TITLE
RFD: AKS & GKE Kubernetes clusters auto-discovery

### PR DESCRIPTION
**Pull request by tigrato**
This PR introduces the RFD specification for extending Kubernetes auto-discovery to Microsoft Azure AKS-managed and Google GCP GKE clusters.

Related to #12048
Part of #13376

### CODE REVIEW

**Review by r0mant — COMMENTED** [review]
@tigrato In general it makes sense, however the RFD is a little hard to follow because it throws a bunch of implementation details and commands at the reader and it's not super clear what the final UX would look like i.e. what kind of set up (if any) will be required from users in order for auto-discovery to run, what kind of set up Teleport will handle automatically, etc.

I propose to refocus the RFD a little bit on the user flows and UX, and either trim implementation details or move them into separate appendices/sections.

For example I would structure the RFD like this:

* Start by outlining that there are 3 configurations AKS cluster can be in and _very briefly_ explain how each of them handles authentication. We don't need details on how you would create a AKS cluster using `az` tool, not at this stage at least.
* For each of the configurations we're basically interested in whether Teleport can autonomously manage access to discovered clusters which leads to these questions:
  * How will Teleport authenticate with the cluster in this case?
  * What are the requirements for Teleport to be able to authenticate with the cluster? I.e. does it need to run with an Azure identity that has some permissions? Is this a one-time setup that can be done with `teleport discover bootstrap` command?
  * What will Teleport do after discovering the cluster?
* Either remove example commands of how you would create a AKS cluster, or move them into a dedicated appendix section so they don't distract the reader from understanding how the actual auto-discovery works.
* Mermaid diagrams are nice but aren't readable unless rendered which makes reviewing them in raw Markdown also distracting. I'd also move them into separate section in appendix.
* Same goes for RBAC policy examples, let's have a dedicated section with all of them explaining "this is the policy Teleport Kube agent needs in order to handle this scenario, it will be setup using teleport discovery bootstrap command".

Basically, I think the ideal structure is to first give the reader a very brief, clear and concise overview of how things will work to help them build a mental model of it, and then gradually expand on each scenario with details.

**Comment by tigrato** [comment]
@r0mant I revised the RFD to include more user-friendly and user-focused details.
